analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.RoslynCfg.cs

```diff
@@ -99,6 +97,15 @@ operation.Kind switch
                             OperationKindEx.Invocation
                                 when IInvocationOperationWrapper.FromOperation(operation) is var invocation && (!invocation.IsVirtual || InstanceReferencesThis(invocation.Instance)) =>
                                 invocation.TargetMethod,
+                            OperationKindEx.Binary
```

Comment:
There should be also cases with OperationKindEx.Unary and examples that cover it.